crates/turborepo-ui/src/tui/task.rs

```diff
@@ -108,32 +108,19 @@ impl TasksByStatus {
     }
 
     pub fn count_all(&self) -> usize {
-        self.task_names_in_displayed_order().len()
+        self.task_names_in_displayed_order().count()
     }
 
-    pub fn task_names_in_displayed_order(&self) -> Vec<String> {
-        let running_names = self
-            .running
-            .iter()
-            .map(|task| task.name().to_string())
-            .collect::<Vec<_>>();
-        let planned_names = self
-            .planned
-            .iter()
-            .map(|task| task.name().to_string())
-            .collect::<Vec<_>>();
-        let finished_names = self
-            .finished
-            .iter()
-            .map(|task| task.name().to_string())
-            .collect::<Vec<_>>();
-
-        [
-            running_names.as_slice(),
-            planned_names.as_slice(),
-            finished_names.as_slice(),
-        ]
-        .concat()
+    pub fn task_names_in_displayed_order(&self) -> impl Iterator<Item = &str> + '_ {
```

Comment:
This change means the function now returns an iterator over all task names instead of an vector of them. Beneficial to us as it results in far fewer allocations (4 vector allocations + 1 allocation per task) to zero allocations.